Firebase/Auth/Source/Public/FIRAuth.h

```diff
@@ -416,26 +416,57 @@ NS_SWIFT_NAME(Auth)
 
 
     @remarks See `FIRAuthErrors` for a list of error codes that are common to all API methods.
-
-    @remarks This method will only exist until the next major Firebase release following 4.x.x.
-        After the next major release the method `signInWithEmail:password:completion:` will support
-        the `FIRAuthDataResultCallback`.
  */
 - (void)signInAndRetrieveDataWithEmail:(NSString *)email
                               password:(NSString *)password
-                            completion:(nullable FIRAuthDataResultCallback)completion;
+                            completion:(nullable FIRAuthDataResultCallback)completion
+                                __attribute__((deprecated));
```

Comment:
If you use 
```
DEPRECATED_MSG_ATTRIBUTE("Please use `signIn(withEmail:password:completion)` instead.")
```
 it should show up in the Xcode warning instead of having to search through the docs.

Reply:
Done